pkg/virt-controller/services/template.go

```diff
@@ -302,6 +304,57 @@ func requestResource(resources *k8sv1.ResourceRequirements, resourceName string)
 	}
 }
 
+func addVhostuserVolume(volumeMounts *[]k8sv1.VolumeMount, volumes *[]k8sv1.Volume) {
+	// "shared-dir" volume name will be used by userspace cni to place the vhostuser socket file
+	*volumeMounts = append(*volumeMounts, k8sv1.VolumeMount{
+		Name:      "shared-dir",
+		MountPath: VhostuserSocketDir,
```

Comment:
@jean-edouard are there any SELinux implications that we need to be paying attention to when introducing shared sockets via volume mounts?

Reply:
Good question. I think that as long as the sockets don't carry critical un-encrypted data, they can be labelled `container_*_t`, without any categories, which would make them usable by any container without upsetting SELinux.
@maiqueb what do you think?

Reply:
+1; not having the categories would simplify this ordeal by a lot ( and that's assumable while these sockets don't carry critical un-encrypted data).

Reply:
@jean-edouard @maiqueb This directory should be accessible for libvirt to create the socket this speecified path `/var/lib/cni/usrcni/`. This socket file should be accessible to the openvswitch.Is ther anything to be handled inside the virt-launcher to allow libvirt to access this directory?

Reply:
I may not have enough context right now, but would it make sense to let virt-handler set up the socket? Is that possible? Using the fsgroup setting may be ok, would have to look deeper.

Reply:
Right, we always have that option - have virt-handler set up the socket correctly. That requires code that will require maintenance. 

The `fsGroup` seems to be a "cheaper" alternative - someone else maintains it. But again - I'm not 100% sure this will work, since it affects the whole pod.

@krsacme you're free to choose from the alternatives above.

Reply:
> would it make sense to let virt-handler set up the socket?

Though `openvswitch` is just a user (expects the file to be created), `libvirt` creates the socket file, so not it is not possible in the current state.

> Could we get away with simply defining the pod security spec's fsGroup attribute ?

Refer the [writeup](https://github.com/krsacme/kubevirt/blob/vhostuser/docs/ovsdpdk.md#userspace-cni]), the socket is created in an `EmtpyDir` volume of the virt-launcher pod. It is then used by the cni, by locally mounting it. 

As part of the PR series, we are configuring libvirt to run with a specific group, which should be enough to create the socket file with the required ownership. I don't think we need to change the pod's group.

Looping @cijohnson as he is also working on the PR.